### PR TITLE
refactor(existing destinations): error handling

### PR DESCRIPTION
**Pull request by Gauravudia**
## Description of the change

> Refactoring error handling for following destinations according to the new error classes introduced by [this](https://github.com/rudderlabs/rudder-transformer/pull/1442) PR :-
- June
- ClickUp
- Refiner
- MailJet
- Mautic

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/mautic/transform.js
Comment (saikumarrs) [inline v0/destinations/mautic/transform.js:42]:
```suggestion
      "Something went wrong while constructing the payload",
```

File: v0/destinations/mailjet/transform.js
Comment (saikumarrs) [inline v0/destinations/mailjet/transform.js:66]:
```suggestion
    `Event type "${messageType}" is not supported`,
```

File: v0/destinations/mailjet/transform.js
Comment (saikumarrs) [inline v0/destinations/mailjet/transform.js:50]:
```suggestion
      "Event type is required",
```

File: v0/destinations/mailjet/transform.js
Comment (saikumarrs) [inline v0/destinations/mailjet/transform.js:31]: Same as below.

File: v0/destinations/june/transform.js
Comment (saikumarrs) [inline v0/destinations/june/transform.js:104]: Same as below.

File: v0/destinations/june/transform.js
Comment (saikumarrs) [inline v0/destinations/june/transform.js:81]: Same as below.

File: v0/destinations/june/transform.js
Comment (saikumarrs) [inline v0/destinations/june/transform.js:30]: Same as below.

File: v0/destinations/clickup/util.js
Comment (saikumarrs) [inline v0/destinations/clickup/util.js:394]:
```suggestion
        "The event was discarded as it was not allow listed in the destination configuration",
```

File: v0/destinations/mautic/utils.js
Comment (saikumarrs) [inline v0/destinations/mautic/utils.js:231]:
```suggestion
      `Failed to fetch contacts: "${JSON.stringify(
        searchContactsResponse.response
      )}"`,
```

**Review by saikumarrs — CHANGES_REQUESTED** [review]
A lot of suggestions are applicable to other places as well. Please address all of them.

File: v0/destinations/mautic/transform.js
Comment (Gauravudia) [reply v0/destinations/mautic/transform.js:90]: Can you confirm if we can `Instrumentation` error type here?

File: v0/destinations/refiner/transform.js
Comment (Gauravudia) [reply v0/destinations/refiner/transform.js]: updated

File: v0/destinations/refiner/transform.js
Comment (Gauravudia) [reply v0/destinations/refiner/transform.js]: updated

File: v0/destinations/refiner/transform.js
Comment (Gauravudia) [reply v0/destinations/refiner/transform.js]: updated

File: v0/destinations/clickup/util.js
Comment (Gauravudia) [reply v0/destinations/clickup/util.js]: updated

File: v0/destinations/mautic/utils.js
Comment (Gauravudia) [reply v0/destinations/mautic/utils.js]: updated

File: v0/destinations/mautic/utils.js
Comment (Gauravudia) [reply v0/destinations/mautic/utils.js]: updated

File: v0/destinations/mautic/utils.js
Comment (Gauravudia) [reply v0/destinations/mautic/utils.js]: updated

File: v0/destinations/mautic/utils.js
Comment (Gauravudia) [reply v0/destinations/mautic/utils.js]: updated

File: v0/destinations/mautic/utils.js
Comment (Gauravudia) [reply v0/destinations/mautic/utils.js]: updated

File: v0/destinations/mautic/transform.js
Comment (Gauravudia) [reply v0/destinations/mautic/transform.js]: updated

File: v0/destinations/mautic/transform.js
Comment (Gauravudia) [reply v0/destinations/mautic/transform.js:224]: stat tags are added

File: v0/destinations/mautic/utils.js
Comment (Gauravudia) [reply v0/destinations/mautic/utils.js]: yes, updated it to configuration

File: v0/destinations/mautic/utils.js
Comment (Gauravudia) [reply v0/destinations/mautic/utils.js]: updated

File: v0/destinations/mautic/utils.js
Comment (Gauravudia) [reply v0/destinations/mautic/utils.js]: updated

File: v0/destinations/mautic/transform.js
Comment (Gauravudia) [reply v0/destinations/mautic/transform.js]: updated

File: v0/destinations/mautic/transform.js
Comment (Gauravudia) [reply v0/destinations/mautic/transform.js]: updated

File: v0/destinations/mautic/transform.js
Comment (Gauravudia) [reply v0/destinations/mautic/transform.js]: updated

File: v0/destinations/mautic/transform.js
Comment (Gauravudia) [reply v0/destinations/mautic/transform.js]: updated

File: v0/destinations/mautic/transform.js
Comment (Gauravudia) [reply v0/destinations/mautic/transform.js]: updated

Comment (Gauravudia) [reply v0/destinations/mautic/transform.js]: Stat tags are added

File: v0/destinations/mautic/transform.js
Comment (Gauravudia) [reply v0/destinations/mautic/transform.js]: updated

File: v0/destinations/mautic/transform.js
Comment (Gauravudia) [reply v0/destinations/mautic/transform.js]: updated

File: v0/destinations/mailjet/transform.js
Comment (Gauravudia) [reply v0/destinations/mailjet/transform.js]: updated

File: v0/destinations/mailjet/transform.js
Comment (Gauravudia) [reply v0/destinations/mailjet/transform.js]: updated

File: v0/destinations/mautic/transform.js
Comment (saikumarrs) [reply v0/destinations/mautic/transform.js]: Yeah, that makes sense.

Comment (Gauravudia) [reply v0/destinations/mautic/transform.js]: Error type updated

**Review by saikumarrs — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

File: v0/destinations/mautic/transform.js
Comment (crownedlake) [inline v0/destinations/mautic/transform.js:280]: a suggestion, if you have multiple errors, you can create one new `errors.ts` file and define all the errors there, import the file and return that error instance from the designated places. It will make your code more readable :)

Comment (Gauravudia) [reply v0/destinations/mautic/transform.js:280]: We have to implement this for all the existing destinations. We can consider this in a separate task.

Comment (yashasvibajpai) [reply v0/destinations/mautic/transform.js:280]: @Gauravudia Lets track this somewhere, if this can be prioritised.

Comment (Gauravudia) [reply v0/destinations/mautic/transform.js:280]: sure @yashasvibajpai

Comment (krishna2020) [reply v0/destinations/mautic/transform.js:280]: @crownedlake still DESTINATION param changes. one more way, we can move few defaults to the error class implementation itself & use Constants for Error messages. Check out this page where we are discussing on these & add your comments - https://www.notion.so/rudderstacks/Transformer-Alerting-Objectives-dcc0dab0bd8640f5bb46d1da45bc69ec

@Gauravudia we can merge this PR & once we finalize the next set of changes, we can pick up. That's the reason to hold off refactoring error classes for next set of destinations.

**Pull request merged**
